modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployService.java

```diff
@@ -654,4 +661,31 @@ public void updateServiceStatus(UUID id, ServiceDeploymentState serviceDeploymen
         deployServiceEntity.setServiceDeploymentState(serviceDeploymentState);
         deployServiceStorage.storeAndFlush(deployServiceEntity);
     }
+
+    /**
+     * Get the migration status of a service by service ID.
+     */
+    public ServiceDeploymentState queryMigrateStateById(String id) {
+        String[] split = id.split("@");
```

Comment:
we must avoid this for sure. 
The migration ID -> can be preferably the ID of the new service or worst case can be a completely new ID